common/isolationgroup/default-isolation-group-state_test.go

```diff
@@ -23,13 +23,110 @@
 package isolationgroup
 
 import (
+	"context"
+	"encoding/json"
+	"errors"
 	"testing"
 
+	"github.com/golang/mock/gomock"
+
+	"github.com/uber/cadence/common/cache"
+	"github.com/uber/cadence/common/dynamicconfig"
+	"github.com/uber/cadence/common/log/loggerimpl"
+	"github.com/uber/cadence/common/persistence"
+
 	"github.com/stretchr/testify/assert"
 
 	"github.com/uber/cadence/common/types"
 )
 
+func TestIsDrainedHandler(t *testing.T) {
+
+	validInput := types.IsolationGroupConfiguration{
+		"zone-1": types.IsolationGroupPartition{
+			Name:  "zone-1",
+			State: types.IsolationGroupStateHealthy,
+		},
+		"zone-2": types.IsolationGroupPartition{
+			Name:  "zone-2",
+			State: types.IsolationGroupStateDrained,
+		},
+	}
+
+	validCfg, _ := mapUpdateGlobalIsolationGroupsRequest(validInput)
```

Comment:
nit: I'd still check these errors with `require.NoError(t, err)`.

Reply:
fair, let me do that